src/System.Text.Json/src/System/Text/Json/JsonElement.cs

```diff
@@ -0,0 +1,458 @@
+
+            throw new FormatException();
+        }
+
+        [CLSCompliant(false)]
+        public bool TryGetUInt64(out ulong value)
+        {
+            CheckValidInstance();
+
+            return _parent.TryGetValue(_idx, out value);
+        }
+
+        [CLSCompliant(false)]
+        public ulong GetUInt64()
+        {
+            if (TryGetUInt64(out ulong value))
+            {
+                return value;
+            }
+
+            throw new FormatException();
+        }
+
+        public bool TryGetDouble(out double value)
+        {
+            CheckValidInstance();
+
+            return _parent.TryGetValue(_idx, out value);
+        }
+
+        public double GetDouble()
+        {
+            if (TryGetDouble(out double value))
+            {
+                return value;
+            }
+
+            throw new FormatException();
+        }
+
+        public bool TryGetSingle(out float value)
+        {
+            CheckValidInstance();
+
+            return _parent.TryGetValue(_idx, out value);
+        }
+
+        public float GetSingle()
+        {
+            if (TryGetSingle(out float value))
+            {
+                return value;
+            }
+
+            throw new FormatException();
+        }
+
+        public bool TryGetDecimal(out decimal value)
+        {
+            CheckValidInstance();
+
+            return _parent.TryGetValue(_idx, out value);
+        }
+
+        public decimal GetDecimal()
+        {
+            if (TryGetDecimal(out decimal value))
+            {
+                return value;
+            }
+
+            throw new FormatException();
+        }
+
+        internal string GetPropertyName()
+        {
+            CheckValidInstance();
+
+            return _parent.GetNameOfPropertyValue(_idx);
+        }
+
+        public string GetRawText()
+        {
+            CheckValidInstance();
+
+            return _parent.GetRawValueAsString(_idx);
+        }
+
+        internal string GetPropertyRawText()
+        {
+            CheckValidInstance();
+
+            return _parent.GetPropertyRawValueAsString(_idx);
+        }
+
+        public ArrayEnumerator EnumerateArray()
+        {
+            CheckValidInstance();
+
+            JsonTokenType tokenType = TokenType;
+
+            if (tokenType != JsonTokenType.StartArray)
+            {
+                throw ThrowHelper.GetJsonElementWrongTypeException(JsonTokenType.StartArray, tokenType);
+            }
+
+            return new ArrayEnumerator(this);
+        }
+
+        public ObjectEnumerator EnumerateObject()
+        {
+            CheckValidInstance();
+
+            JsonTokenType tokenType = TokenType;
+
+            if (tokenType != JsonTokenType.StartObject)
+            {
+                throw ThrowHelper.GetJsonElementWrongTypeException(JsonTokenType.StartObject, tokenType);
+            }
+
+            return new ObjectEnumerator(this);
+        }
+
+        public override string ToString()
```

Comment:
Do we need to add a `[DebuggerDisplay("{ToString(),raw}")]` attribute to this type or will the debugger show the ToString value by default?

Similarly, what is the most useful info to show in the debugger on an instance of JsonDocument/JsonProperty/etc.

Reply:
The debugger defaults to showing the ToString value.